### DISCUSSION TITLE
Redesign of contact screen

### DISCUSSION THREAD

**Discussion by ehuelsmann** [ledgersmb/LedgerSMB#5823] [Design]
There's been talk of creating a more intuitive / self-documenting contact screen. Some time ago, I've created a design that I think will help people understand the structure of contacts better. Attaching it here for others to see and for reference

![customer-and-accounts-structure-proposal](https://cloud.githubusercontent.com/assets/2326559/10563539/3521670c-758e-11e5-9239-ce1668c42670.PNG)

**Comment by ehuelsmann** [comment]
Looking at this screen design, I'm now thinking that the customer classes "Hot lead" and "Cold lead" are really not customer classes, but customer states, like "active"(having a begin date, but no end date, or an end date in the future) and "deleted"(having an end date in the past)

**Comment by ehuelsmann** [comment]
For anybody wanting to create variants of the picture above, here's the spreadsheet I used to "draw" it and made a screenshot from.

[Company-and-accounts-structure-proposal.xlsx](https://github.com/ledgersmb/LedgerSMB/files/15570/Company-and-accounts-structure-proposal.xlsx)

**Comment by einhverfr** [comment]
I like how it shows the relationships between things.  That is something that needs to be improved.

**Comment by ehuelsmann** [comment]
Note that the "Class" `Vendor` bit is not helpful here and should be left out: a company can have lots of roles, including Vendor and Customer, but those are modelled through "Accounts".
